pkg/utils/nodepool/nodepool.go

```diff
@@ -76,15 +87,28 @@ func ListManaged(ctx context.Context, c client.Client, cloudProvider cloudprovid
 	}), nil
 }
 
-func NodeClaimEventHandler() handler.EventHandler {
-	return handler.EnqueueRequestsFromMapFunc(func(_ context.Context, o client.Object) []reconcile.Request {
+func NodeClaimEventHandler(c client.Client, staticOnly bool) handler.EventHandler {
```

Comment:
Is this better handled with the `...options` variadic kind of handling? That way that callers that don't care about specifying False for this every time don't have to handle passing that in